src/java.desktop/share/native/libfontmanager/freetypeScaler.c

```diff
@@ -1023,22 +1304,26 @@ static jlong
         }
     }
 
-    if (context->fmType == TEXT_FM_ON) {
+    if (context->fmType == TEXT_FM_ON && outlineGlyph) {
         float advh = FTFixedToFloat(ftglyph->linearHoriAdvance);
         glyphInfo->advanceX =
             (float) (advh * FTFixedToFloat(context->transform.xx));
         glyphInfo->advanceY =
             (float) - (advh * FTFixedToFloat(context->transform.yx));
     } else {
         if (!ftglyph->advance.y) {
-            glyphInfo->advanceX = FT26Dot6ToFloat(ftglyph->advance.x);
+            glyphInfo->advanceX = FT26Dot6ToFloat(
+                    FT_MulFix(ftglyph->advance.x, manualScale));
```

Comment:
This isn't right. In this non-FM code path we are taking the pixel advance of the glyph which should be a whole integer even if we store it in a float. 
Applying this manualScale breaks that, even if it is just for the emoji fonts.

I think you need to do whatever is necessary to round this here - probably to the nearest whole pixel - for the emoji case.